modules/react/collection/lib/useBaseListModel.tsx

```diff
@@ -5,11 +5,25 @@ import {useUniqueId, createModelHook, Generic} from '@workday/canvas-kit-react/c
 
 export type Orientation = 'horizontal' | 'vertical';
 
-export const defaultGetId = (item: any): string => {
+export const defaultGetId = (item: Generic): string => {
+  if (process.env.NODE_ENV === 'development') {
+    if (typeof item === 'object' && item.id === undefined) {
+      console.warn(
+        "List item was an object, but no `getId` was passed to the model to inform the list where to find the item's identifier. Please pass a `getId` to the list model"
+      );
+    }
+  }
   return typeof item === 'string' ? item : item.id || '';
```

Comment:
Looks like this still needs the check for `undefined`? When logging out `item` for some reason in the autocomplete example, there's an instance where the item can be `undefined,

Updating the return to:
```
return item === undefined ? '' : typeof item === 'string' ? item : item.id || '';
```
Fixed the issue for both `defaultGetId` and `defaultGetTextValue`